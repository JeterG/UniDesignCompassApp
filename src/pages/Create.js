import React from "react";
import inter from "../images/home-image-min.jpg";
import Layout from '../components/layout';
import { CardDeck, Card, Button, Form } from 'react-bootstrap';
import CompassForm from '../components/compassForm'

const CustomOrPremade = (props) =>  (
    <Form.Group controlId="formBasicEmail">
      <Form.Label>Do you want a custom Compass or a prebuilt Compass?</Form.Label>
      <Button variant="primary" > Custom </Button>
      <Button variant="primary" > Default </Button>
    </Form.Group>
)

const PhaseNumber = (props) =>  (
  <Form.Group controlId="exampleForm.ControlSelect1">
    <Form.Label>How many phases are in your design process</Form.Label>
    <Form.Control as="select">
      <option>1</option>
      <option>2</option>
      <option>3</option>
      <option>4</option>
      <option>5</option>
      <option>7</option>
      <option>8</option>
      <option>9</option>
      <option>10</option>
      <option>11</option>
      <option>12</option>
    </Form.Control>
  </Form.Group>
)

const DescribePhases = (props) =>  (
  <Form.Group controlId="exampleForm.ControlSelect1">
    <Form.Label>Describe each phase in your design process</Form.Label>
      <Form.Label>Title of Phase</Form.Label>
      <Form.Control type="email" placeholder="Title of Title" />
      <Form.Label>Description of your Phase</Form.Label>
      <Form.Control as="textarea" rows="3" />
  </Form.Group>
)

const CreatePage = (props) => (
  <Layout>
     <CompassForm>
      <CustomOrPremade />
     </CompassForm>
     <CompassForm>
      <PhaseNumber/>
     </CompassForm>
     <CompassForm>
      <DescribePhases/>
     </CompassForm>
  </Layout>
);
export default CreatePage;
