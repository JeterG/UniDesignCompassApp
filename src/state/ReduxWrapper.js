import React from 'react';
import { Provider } from 'react-redux';
import createStore from "./store";
import { PersistGate } from 'redux-persist/integration/react'

export default ({ element }) => (
  <Provider store={createStore().store}>
    <PersistGate loading={null} persistor={createStore().persistor}>
      {element}
    </PersistGate>
  </Provider>
);