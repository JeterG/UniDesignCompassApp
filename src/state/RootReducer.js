import { combineReducers } from 'redux';
import Reducer from './reducers';

export default combineReducers({ state: Reducer });
