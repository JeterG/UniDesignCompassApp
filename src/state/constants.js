export const AUTH_USER = "AUTH_USER";
export const UPDATE_USER = "UPDATE_USER";
export const LOGOUT_USER = "LOGOUT_USER";
