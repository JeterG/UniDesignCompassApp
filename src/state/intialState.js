export const initialState = {
  user : {},
  isAuthenticated: false,
}