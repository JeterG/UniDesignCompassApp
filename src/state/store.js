import {  createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import RootReducer from './RootReducer';

const composeEnhancers = (typeof window !== `undefined`) ? (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose ) : compose;

export default function configureStore() {
  return createStore(
    RootReducer,
    composeEnhancers(applyMiddleware(thunk))
  );
 }