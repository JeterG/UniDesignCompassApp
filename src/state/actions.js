import { FETCH_USER_INFO, AUTH_USER } from "./constants"

export const fetchUserInfo = (orderID) => ({
  type: FETCH_USER_INFO,
  payload: orderID
})

export const authenticateUser = (authenticated = false) => dispatch => {

  dispatch(
    {
      type: AUTH_USER,
      payload: "true",
      aws_user: [1]
    }
  )
}
