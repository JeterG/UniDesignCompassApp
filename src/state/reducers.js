import {initialState} from "./intialState"
import {AUTH_USER} from "./constants"

export default (state = initialState, action) => {
  switch(action.type) {
    case AUTH_USER :
      return {
        ...state,
        user: action.payload,
        isAuthenticated: true
      };
    default:
      return state;
  }
}

