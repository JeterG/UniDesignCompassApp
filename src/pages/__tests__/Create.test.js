import React from "react"
import renderer from "react-test-renderer"
import { store } from "../../state/store";
import { Provider } from 'react-redux';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import {CreatePage, DescribePhase, PhaseNumber, ChooseStructure, SubmitCompass } from '../Create';
configure({adapter:new Adapter()});

// const tree = shallow(<Provider store={store}><CreatePage /></Provider>).dive().dive();

describe("Testing Create Page",()=>{

    const DescribePhaseComponent = shallow(<DescribePhase debug/>)
    it("Check DescribePhase Render",()=>{
        expect(shallowToJson(DescribePhaseComponent)).toMatchSnapshot();
    });

    const PhaseNumberComponent = shallow(<PhaseNumber debug/>)
    it("Check PhaseNumber Render",()=>{
        expect(shallowToJson(PhaseNumberComponent)).toMatchSnapshot();
    });

    const ChooseStructureComponent = shallow(<ChooseStructure debug/>)
    it("Check ChooseStructure Render",()=>{
        expect(shallowToJson(ChooseStructureComponent)).toMatchSnapshot();
    });

    const SubmitCompassComponent = shallow(<SubmitCompass debug/>)
    it("Check SubmitCompass Render",()=>{
        expect(shallowToJson(SubmitCompassComponent)).toMatchSnapshot();
    });

    const CreatePageComponent = shallow(<CreatePage debug/>)
    it("Check DescribePhase Render",()=>{
        expect(shallowToJson(CreatePageComponent)).toMatchSnapshot();
    });

    // it("Check Components",()=>{
    //     expect(tree.find('Button')).toHaveLength(2);
    // })

})