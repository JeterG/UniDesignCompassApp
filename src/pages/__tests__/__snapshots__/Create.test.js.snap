// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Create Page Check ChooseStructure Render 1`] = `
<FormGroup
  as="div"
  controlId="formBasicEmail"
>
  <FormLabel
    className="form-title"
    column={false}
    srOnly={false}
  >
    Choose a Compass Style:
  </FormLabel>
  <div
    className="input-area"
  >
    <Button
      active={false}
      className=""
      disabled={false}
      id="Custom"
      onClick={[Function]}
      type="button"
      variant="primary"
    >
       Custom 
    </Button>
    <Button
      active={false}
      className="input-button"
      disabled={false}
      id="Default"
      onClick={[Function]}
      type="button"
      variant="primary"
    >
       Default 
    </Button>
  </div>
</FormGroup>
`;

exports[`Testing Create Page Check DescribePhase Render 1`] = `
<FormGroup
  as="div"
  controlId="exampleForm.ControlSelect1"
>
  <FormLabel
    className="form-sub-title"
    column={false}
    srOnly={false}
  >
    Phase 
    NaN
  </FormLabel>
  <FormControl
    as="input"
    name="title"
    onChange={[Function]}
    placeholder="Title"
    type="text"
    value=""
  />
  <FormLabel
    column={false}
    srOnly={false}
  >
    Description of your Phase
  </FormLabel>
  <FormControl
    as="textarea"
    name="description"
    onChange={[Function]}
    placeholder="Description"
    rows="3"
    value=""
  />
</FormGroup>
`;

exports[`Testing Create Page Check DescribePhase Render 2`] = `
<Layout>
  <h2>
     Create Your Compass 
  </h2>
  <div
    className="form-container"
  >
    <ChooseStructure
      handleCompassType={[Function]}
    />
  </div>
</Layout>
`;

exports[`Testing Create Page Check PhaseNumber Render 1`] = `
<FormGroup
  as="div"
  controlId="exampleForm.ControlSelect1"
>
  <FormLabel
    className="form-title"
    column={false}
    srOnly={false}
  >
    How many phases are in your design process
  </FormLabel>
  <div
    className="input-area"
  >
    <FormControl
      as="select"
    >
      <option
        value="0"
      >
        -
      </option>
      <option
        value="2"
      >
        2
      </option>
      <option
        value="3"
      >
        3
      </option>
      <option
        value="4"
      >
        4
      </option>
      <option
        value="5"
      >
        5
      </option>
      <option
        value="6"
      >
        6
      </option>
      <option
        value="7"
      >
        7
      </option>
      <option
        value="8"
      >
        8
      </option>
      <option
        value="9"
      >
        9
      </option>
      <option
        value="10"
      >
        10
      </option>
      <option
        value="11"
      >
        11
      </option>
      <option
        value="12"
      >
        12
      </option>
    </FormControl>
  </div>
</FormGroup>
`;

exports[`Testing Create Page Check SubmitCompass Render 1`] = `
<FormGroup
  as="div"
  controlId="formBasicEmail"
>
  <FormLabel
    className="form-title"
    column={false}
    srOnly={false}
  >
    Title of your Design Compass
  </FormLabel>
  <div
    className="input-area"
  >
    <FormControl
      as="input"
      name="title"
      placeholder="Title"
      type="text"
    />
    <Button
      active={false}
      className="input-button"
      disabled={false}
      type="button"
      variant="primary"
    >
       Submit Compass 
    </Button>
  </div>
</FormGroup>
`;
